src/scheduler/misc.c

```diff
@@ -1701,3 +1701,44 @@ res_to_str_re(void *p, enum resource_fields fld, char **buf,
 		return "";
 	return *buf;
 }
+
+/**
+ * @brief
+ *	Helper function used to copy the given source string to the destination string. It also
+ *	frees the destination and then allocates required memory before copying.
+ *
+ * @param[in] dest - Address of pointer to destination string
+ * @param[in] dest - pointer to source string
+ *
+ * @retval
+ * @return 0 - Failure
+ * @return 1 - Success
+ *
+ * @par Side Effects:
+ *	None
+ *
+ *
+ */
+int
+copy_attr_value(char **dest, char *src)
+{
+	int ret = 0;
+
+	if (*dest != NULL)
+		free(*dest);
```

Comment:
@arungrover 
Instead of using free I am now using realloc() as per the earlier comment from Bhroam. So I don't think the problem you mentioned will arise now.